lib/scripts/diffExperimentBuilds.py

```diff
@@ -0,0 +1,255 @@
+            spack_setup_script = f"{name}/wkp/setup.sh"
+            # Define the ramble command
+            ramble_command = f"{name}/wkp/ramble/bin/ramble --workspace-dir {name}/wkp/{name}/{exper}/{name}/{cluster}/workspace workspace setup"
+            # Combine sourcing the script and running the command
+            run_str = f"bash -c 'source {spack_setup_script} && {ramble_command}"
+            if args.run_experiment:
+                run_str += f" && {name}/wkp/ramble/bin/ramble --workspace-dir {name}/wkp/{name}/{exper}/{name}/{cluster}/workspace on"
+            run_str += "'"
+            subprocess.run(
+                run_str,
+                shell=True,
+                check=True,
+                text=True,
+            )
+            # Run the `spack find` command to get the hash
+            pkg_hash = subprocess.run(
+                [f"{name}/wkp/spack/bin/spack", "find", "--hash", exper],
+                text=True,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,  # Capture errors
+            )
+            spec_result = subprocess.run(
+                [
+                    f"{name}/wkp/spack/bin/spack",
+                    "spec",
+                    "--yaml",
+                    pkg_hash.stdout.strip(),
+                ],
+                text=True,
+                stdout=subprocess.PIPE,
+            )
+            # Write the output to a file
+            yaml_file_path = f"{name}-{exper}.yaml"
+            if os.path.isfile(yaml_file_path):
+                os.remove(yaml_file_path)
+            with open(yaml_file_path, "w") as yaml_file:
+                yaml_file.write(spec_result.stdout)
+
+    for exper in experiments.keys():
+        old_file = f"./{old_name}-{exper}.yaml"
+        new_file = f"./{new_name}-{exper}.yaml"
+
+        # Path to the Spack setup script
+        spack_setup_script = f"{name}/wkp/setup.sh"
+        # Define the ramble command
+        cmd = (
+            f"{old_name}/wkp/spack/bin/spack-python altdiff.py -t {old_file} {new_file}"
+        )
+        # Combine sourcing the script and running the command
+        try:
+            diff = subprocess.run(
+                f"bash -c 'source {spack_setup_script} && {cmd}'",
+                shell=True,
+                check=True,
+                text=True,
+                stdout=subprocess.PIPE,
+            )
+
+            diff_str = diff.stdout
+
+            if "DifferentSpecs=True" in diff_str:
+                color.cprint(f"@*rThe specs for {exper} are different.@.")
+            elif "DifferentSpecs=False" in diff_str:
+                color.cprint(f"@*gThe specs for {exper} are the same.@.")
+            else:
+                raise ValueError("Expected value in output")
+            print(diff_str)
+        except:  # noqa
+            color.cprint("@*o" + f"The specs for {exper} could not be compared.@.")
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
Please add docs for this script.

Reply:
Docs page for scripts added in [c2605fa](https://github.com/LLNL/benchpark/pull/655/commits/c2605fa3c5b71db4b5efe80d57d79ca8b20478b1)